src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Options/PackageSourcesPage.cs

```diff
@@ -0,0 +1,243 @@
+                _suppressSettingValuesChanged = true;
+
+                switch (moniker)
+                {
+                    case MonikerPackageSources:
+                        return await Task.Run(
+                            () => SavePackageSources<T>(packageSourcesList, cancellationToken),
+                            cancellationToken);
+
+                    case MonikerMachineWideSources:
+                        return await Task.Run(
+                            () => SetIsEnabledOnMachineWidePackageSources(packageSourcesList, cancellationToken),
+                            cancellationToken);
+
+                    default:
+                        throw new InvalidOperationException();
+                }
+            }
+            finally
+            {
+                // Resume listening to setting changes after saving.
+                _suppressSettingValuesChanged = false;
+            }
+        }
+
+        private ExternalSettingOperationResult SetIsEnabledOnMachineWidePackageSources(
+            IList<IDictionary<string, object>> packageSourcesList,
+            CancellationToken cancellationToken)
+        {
+            ExternalSettingOperationResult result;
+
+            try
+            {
+                var machineWidePackageSources = LoadPackageSources(isMachineWide: true);
+
+                foreach (PackageSource originalMachineWideSource in machineWidePackageSources)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    string originalPackageSourceName = originalMachineWideSource.Name;
+                    IDictionary<string, object> targetPackageSource = packageSourcesList
+                        .Single(packageSourceDictionary =>
+                            packageSourceDictionary[MonikerSourceName].ToString() == originalPackageSourceName);
+
+                    bool originalIsEnabled = originalMachineWideSource.IsEnabled;
+                    bool targetIsEnabled = (bool)targetPackageSource[MonikerIsEnabled];
+
+                    if (originalIsEnabled != targetIsEnabled)
+                    {
+                        if (targetIsEnabled)
+                        {
+                            _packageSourceProvider.EnablePackageSource(originalPackageSourceName);
+                        }
+                        else
+                        {
+                            _packageSourceProvider.DisablePackageSource(originalPackageSourceName);
+                        }
+                    }
+
+                    // Only one value can be set at a time from Unified Settings, so we're done once a change is made.
+                    break;
+                }
+
+                result = ExternalSettingOperationResult.Success.Instance;
+            }
+            catch (OperationCanceledException ex) when (cancellationToken.IsCancellationRequested)
+            {
+                throw ex;
```

Comment:
You mean "throw;" not "throw ex;".  But it seems like this could be avoided anyway, as you could just add an exception filter to the general "catch (Exception)" clause below to prevent it from catching cancellation exceptions.

Reply:
Same comment on the next method.